In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
path = "https://raw.githubusercontent.com/jbrownlee/Datasets/master/wheat-seeds.csv"

In [32]:
headernames = ['seed 1','seed 2','seed 3','seed 4','seed 5','seed 6','seed 7','class']

In [34]:
data = pd.read_csv(path,names = headernames)

In [35]:
data

seed 1  seed 2  seed 3  seed 4  seed 5  seed 6  seed 7  class
0     15.26   14.84  0.8710   5.763   3.312   2.221   5.220      1
1     14.88   14.57  0.8811   5.554   3.333   1.018   4.956      1
2     14.29   14.09  0.9050   5.291   3.337   2.699   4.825      1
3     13.84   13.94  0.8955   5.324   3.379   2.259   4.805      1
4     16.14   14.99  0.9034   5.658   3.562   1.355   5.175      1
..      ...     ...     ...     ...     ...     ...     ...    ...
205   12.19   13.20  0.8783   5.137   2.981   3.631   4.870      3
206   11.23   12.88  0.8511   5.140   2.795   4.325   5.003      3
207   13.20   13.66  0.8883   5.236   3.232   8.315   5.056      3
208   11.84   13.21  0.8521   5.175   2.836   3.598   5.044      3
209   12.30   13.34  0.8684   5.243   2.974   5.637   5.063      3

[210 rows x 8 columns]

In [36]:
data.shape

(210, 8)

In [37]:
data.head()

seed 1  seed 2  seed 3  seed 4  seed 5  seed 6  seed 7  class
0   15.26   14.84  0.8710   5.763   3.312   2.221   5.220      1
1   14.88   14.57  0.8811   5.554   3.333   1.018   4.956      1
2   14.29   14.09  0.9050   5.291   3.337   2.699   4.825      1
3   13.84   13.94  0.8955   5.324   3.379   2.259   4.805      1
4   16.14   14.99  0.9034   5.658   3.562   1.355   5.175      1

let's Implement the basic step of data preprocessing

In [11]:
X = data.iloc[:,:-1].values
y = data.iloc[:,-1].values

Here we are dividing the data into train and test splits, with 70% of the training data and 30% of testing data

In [26]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X,y, test_size=0.20)

let's do the data scaling

In [27]:
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()
scaler.fit(X_train)
X_train = scaler.transform(X_train)
X_test = scaler.transform(X_test)

Now we are training the model with the help of the GaussianNB class of sklearn as follows

In [28]:
from sklearn.naive_bayes import GaussianNB
clf = GaussianNB()
clf.fit(X_train, y_train)

GaussianNB()

Now, we have to make the prediction

In [29]:
y_pred = clf.predict(X_test)

In [30]:
from sklearn.metrics import classification_report, confusion_matrix, accuracy_score
result = confusion_matrix(y_test, y_pred)
print("Confusion Matrix:")
print(result)

Confusion Matrix:
[[12  0  2]
 [ 0 13  0]
 [ 0  0 15]]


In [31]:
result1 = classification_report(y_test, y_pred)
print("Classification Report:",)
print (result1)
result2 = accuracy_score(y_test,y_pred)
print("Accuracy:",result2)

Classification Report:
              precision    recall  f1-score   support

           1       1.00      0.86      0.92        14
           2       1.00      1.00      1.00        13
           3       0.88      1.00      0.94        15

    accuracy                           0.95        42
   macro avg       0.96      0.95      0.95        42
weighted avg       0.96      0.95      0.95        42

Accuracy: 0.9523809523809523


**Description about the report**


*As the testing percentage is 20, 42 samples were participated in testing phase out of 210.In that 42, class1 = 14,
            class2 = 13,
            class3 = 15,
from the confusion matrix we can conclude that out of 14 class1 samples, 12 are correctly predicted and 2 are wrongly predicted as class 3 .

Out of 13 class2 samples, all are correctly predicted.

Out of 15 class3 samples, all are correctly predicted.*

*The ACCURACY of this model on Wheat seeds data set using Navie Bayes classifier is 95.23%*